virtualizarr/manifests/manifest.py

```diff
@@ -54,6 +65,110 @@ def with_validation(
         return ChunkEntry(path=path, offset=offset, length=length)
 
 
+class ManifestStore(Store):
+    supports_writes: bool = False
+    supports_deletes: bool = False
+    supports_partial_writes: bool = False
+    supports_listing: bool = True
+
+    fs: AsyncFileSystem
+    vds: xr.Dataset
```

Comment:
Technically this could also be a `DataTree`, or maybe a dict of `Datasets`. But we might never need that.